extension/src/paymentHandler/payment-handler.js

```diff
@@ -39,9 +41,31 @@ function _getPaymentHandlers (paymentObject) {
     handlers.push(getPaymentMethodsHandler)
   if (paymentObject.custom.fields.makePaymentRequest && !paymentObject.custom.fields.makePaymentResponse)
     handlers.push(makePaymentHandler)
+  if (paymentObject.custom.fields.makePaymentResponse
+    && paymentObject.custom.fields.submitAdditionalPaymentDetailsRequest
+    && isSubmitAdditionalPaymentDetailsRequestNew(paymentObject))
+    handlers.push(submitPaymentDetailsHandler)
   return handlers
 }
 
+function isSubmitAdditionalPaymentDetailsRequestNew (paymentObject) {
+  const interfaceInteraction = paymentObject.interfaceInteractions
+    .find(interaction => interaction.fields.type === CTP_INTERACTION_TYPE_SUBMIT_PAYMENT_DETAILS)
+  if (interfaceInteraction) {
+    const oldSubmitDetailsRequest = interfaceInteraction.fields.request
+    if (oldSubmitDetailsRequest) {
+      const oldSubmitDetailsRequestObj = JSON.parse(oldSubmitDetailsRequest)
+      const newSubmitDetailsRequestObj = JSON.parse(paymentObject.custom.fields.submitAdditionalPaymentDetailsRequest)
```

Comment:
could you please add JSON validation checks into validateRequestFields which is missing for additional details requests, we can't throw exceptions in here because of our design, and make sure we have a test for those/ you could update existing tests.

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/271/commits/418d750d3d38a67a4428b7fc2094207e84f09996